examples/manipulation-demo-streamlit.py

```diff
@@ -0,0 +1,112 @@
+from rai.tools.ros.native import GetCameraImage, Ros2GetTopicsNamesAndTypesTool
+from rai.utils.model_initialization import get_llm_model
+
+
+@st.cache_resource
+def initialize_graph():
+    rclpy.init()
+    node = RaiBaseNode(node_name="manipulation_demo")
+    node.declare_parameter("conversion_ratio", 1.0)
+    node.qos_profile.reliability = rclpy.qos.ReliabilityPolicy.RELIABLE
+
+    tools = [
+        GetObjectPositionsTool(
+            node=node,
+            target_frame="panda_link0",
+            source_frame="RGBDCamera5",
+            camera_topic="/color_image5",
+            depth_topic="/depth_image5",
+            camera_info_topic="/color_camera_info5",
+        ),
+        MoveToPointTool(node=node, manipulator_frame="panda_link0"),
+        GetCameraImage(node=node),
+        Ros2GetTopicsNamesAndTypesTool(node=node),
+    ]
+
+    system_prompt = """
+    You are a robotic arm with interfaces to detect and manipulate objects.
+    Here are the coordinates information:
+    x - front to back (positive is forward)
+    y - left to right (positive is right)
+    z - up to down (positive is up)
+
+    Before starting the task, make sure to grab the camera image to understand the environment.
+    """
+
+    llm = get_llm_model(model_type="complex_model", streaming=True)
+    agent = create_conversational_agent(
+        llm=llm,
+        tools=tools,
+        system_prompt=system_prompt,
+    )
+    return agent
+
+
+def main():
+    st.set_page_config(
+        page_title="RAI Manipulation Demo",
+        page_icon=":robot:",
+    )
+    st.title("RAI Manipulation Demo")
+    st.markdown("---")
+
+    st.sidebar.header("Tool Calls History")
+
+    if "graph" not in st.session_state:
+        graph = initialize_graph()
+        st.session_state["graph"] = graph
+
+    if "messages" not in st.session_state:
+        st.session_state["messages"] = [
+            AIMessage(content="Hi! I am a robotic arm. What can I do for you?")
+        ]
+
+    prompt = st.chat_input()
+    for msg in st.session_state.messages:
+        if isinstance(msg, AIMessage):
+            if msg.content:
+                st.chat_message("assistant").write(msg.content)
+        elif isinstance(msg, HumanMultimodalMessage):
+            continue
+        elif isinstance(msg, HumanMessage):
+            st.chat_message("user").write(msg.content)
+        elif isinstance(msg, ToolMessage):
+            with st.sidebar.expander(f"Tool: {msg.name}", expanded=False):
+                st.code(msg.content, language="json")
+
+    if prompt:
+        st.session_state.messages.append(HumanMessage(content=prompt))
+        st.chat_message("user").write(prompt)
+        with st.chat_message("assistant"):
+            st_callback = get_streamlit_cb(st.container())
+            streamlit_invoke(
+                st.session_state["graph"], st.session_state.messages, [st_callback]
+            )
```

Comment:
@knicked 
Not sure if this should be addressed under this PR, but I noticed that camera image is not shown in streamlit:
![image](https://github.com/user-attachments/assets/e706fc29-5605-4b98-aa7a-2d52c3001380)

Reply:
This PR uses the currently existing streamlit integration with the llm agent, which apparently does not support displaying images. I will add this functionality in a separate PR.